init/gneiss-broker-message.adb

```diff
@@ -1,14 +1,51 @@
 
-with Gneiss.Main;
 with Gneiss.Broker.Lookup;
-with Gneiss.Protocol;
+with Gneiss.Packet;
 with Gneiss_Log;
+with Gneiss_Internal;
 
 package body Gneiss.Broker.Message with
    SPARK_Mode
 is
 
-   package Proto is new Gneiss.Protocol (RFLX.Types.Byte, RFLX_String);
+   function Image (V : RFLX.Session.Kind_Type) return String is
+      (case V is
+         when RFLX.Session.Message => "Message",
+         when RFLX.Session.Log     => "Log",
+         when RFLX.Session.Memory  => "Memory",
+         when RFLX.Session.Rom     => "Rom",
+         when RFLX.Session.Timer   => "Timer");
+
+   procedure Setup_Service (State : in out Broker_State;
+                            Kind  :        RFLX.Session.Kind_Type;
+                            Index :        Positive);
+
+   procedure Setup_Service (State : in out Broker_State;
+                            Kind  :        RFLX.Session.Kind_Type;
+                            Index :        Positive)
+   is
+      Ignore_Success : Integer;
+   begin
+      if
+         State.Components (Index).Serv (Kind).Broker > -1
+         and then State.Components (Index).Serv (Kind).Disp > -1
+      then
+         return;
+      end if;
+      Gneiss_Syscall.Socketpair (State.Components (Index).Serv (Kind).Broker,
+                                 State.Components (Index).Serv (Kind).Disp);
+      if
+         State.Components (Index).Serv (Kind).Broker < 0
+         or else State.Components (Index).Serv (Kind).Disp < 0
+      then
+         Gneiss_Log.Error ("Failed to create service fds");
+         Gneiss_Syscall.Close (State.Components (Index).Serv (Kind).Broker);
+         Gneiss_Syscall.Close (State.Components (Index).Serv (Kind).Disp);
```

Comment:
In this case `close` will return an `EBADF` error since `-1` is a bad file descriptor. The same would happen if we inserted a file descriptor with a valid number, that has been closed before. Since we do not check errors in `close` (if the file descriptor is invalid it basically is already closed, so the implicit post condition is already valid) nothing bad happens if it returns an error. So the worst outcome is a warning message on stderr.
We should at least decide if we want to always or never check.

Reply:
According to the man page of `close()` it is not wise to retry close on error and the purpose of the return value is diagnostics and to be aware that a previous operation was unsuccessful (e.g. on NFS you may only realize that a previous write failed and you file was not properly persisted when writing it - in that case you could write the whole file again).

Hence, I'd leave our `Close` interface as is an never check any errors.